File: tcpclient/src/main/scala/com/pointr/tcp/sparcle/AppContainerServerIf.scala
package com.pointr.tcp.sparcle

import com.pointr.tcp.rpc.{P2pReq, P2pResp, ServerIf, ServerIfConf}
import com.pointr.tcp.util.Logger

class AppContainerServerIf(conf: ServerIfConf) extends ServerIf("AppContainerServer", Option(conf)) with Logger {

  import AppContainerService._

  override def service(req: P2pReq[_]): P2pResp[_] = {
    debug(s"Received request ${req.toString}")
    req match {
      case o: ExecuteAppCommandReq =>
        ExecuteAppCommandResp(s"YogiBearAndBooBooBearCommandResponse to ${o.toString}")
      case o: ExecuteAppCommand2Req =>
        ExecuteAppCommand2Resp(ExecuteAppCommand2Struct(s"YogiBearAndBooBooBearCommand2 response to ${o.toString}", "GPU123"))
      case o: GetAppStatusReq =>
        GetAppStatusResp(s"GPU123 response to ${o.toString}")
      case o: ReadNextFromSparcleReq =>
        ReadNextFromSparcleResp(s"GPU123 response to ${o.toString}")
      case o: WriteAppDataToSparcleReq =>
        WriteAppDataToSparcleResp(s"OK response to ${o.toString}")
      case o: WriteFeedDataToSparcleReq =>
        WriteFeedDataToSparcleResp(s"OK response to ${o.toString}")
      case o: WriteAppNativeDataToSparcleReq =>
        WriteAppNativeDataToSparcleResp(s"SUCCESS response to ${o.toString}")
      case _ => throw new IllegalArgumentException(s"Unknown service type ${req.getClass.getName}")
    }
  }
}

File: tcpclient/src/main/java/com/pointr/tcp/java/sparcle/AppContainerServerIf.java
package com.pointr.tcp.java.sparcle;
import com.pointr.tcp.rpc.*;
import com.pointr.tcp.rpc.ServerIf;

import com.pointr.tcp.java.sparcle.AppContainerService;
import com.pointr.tcp.java.sparcle.AppContainerService.*;
import com.pointr.tcp.sparcle.AppContainerService.WriteAppDataToSparcleStruct;
import com.pointr.tcp.sparcle.AppContainerService.WriteFeedDataToSparcleStruct;
import com.pointr.tcp.sparcle.AppContainerService.WriteAppNativeDataToSparcleStruct;
import scala.Option;

public class AppContainerServerIf extends ServerIf {

  public AppContainerServerIf(String serverIfName, Option<ServerIfConf> confOpt) {

    super(serverIfName, confOpt);
  }

  @Override
  public P2pResp<?> service(P2pReq<?> req) {
    if (req instanceof AppContainerService.ExecuteAppCommandReq) {
      return new ExecuteAppCommandResp(String.format("YogiBearAndBooBooBearCommandResponse to %s",req.toString()));
    } else if (req instanceof AppContainerService.ExecuteAppCommand2Req) {
      return new ExecuteAppCommand2Resp(new ExecuteAppCommand2JavaStruct(String.format("YogiBearAndBooBooBearCommand2Response to %s",req.toString()),"GPU123"));
    }  else if (req instanceof AppContainerService.GetAppStatusReq) {
      return new GetAppStatusResp(String.format("GPU123 Response to %s",req.toString()));
    }   else if (req instanceof AppContainerService.IsAppRunningReq) {
      return new IsAppRunningResp(true);
    } else if (req instanceof AppContainerService.ReadNextFromSparcleReq) {
      return new ReadNextFromSparcleResp(String.format("ReadNext Resp to %s",req.toString()));
    } else if (req instanceof AppContainerService.WriteAppDataToSparcleReq) {
      return new WriteAppDataToSparcleResp(new WriteAppDataToSparcleStruct("GPURefid",String.format("AppData Response to %s",req.toString()).getBytes()));
    } else if (req instanceof AppContainerService.WriteFeedDataToSparcleReq) {
      return new WriteFeedDataToSparcleResp(new WriteFeedDataToSparcleStruct("GPURefid",String.format("Feed Response to %s",req.toString()).getBytes()));
    } else if (req instanceof AppContainerService.WriteAppNativeDataToSparcleReq) {
      return new WriteAppNativeDataToSparcleResp(new WriteAppNativeDataToSparcleStruct("GPURefid",
          String.format("AppNativeData Response to %s",req.toString()).getBytes()));
    } else {
      throw new UnsupportedOperationException("What type of request is this: %s".format(req.toString()));
    }
  }
}
